src/ai/backend/manager/idle.py

```diff
@@ -302,22 +321,61 @@ async def check_session(self, session: Row, dbconn: SAConnection) -> bool:
 
 class UtilizationIdleChecker(BaseIdleChecker):
     """
-    Checks the idleness of a session by the current utilization of all
-    compute devices and agents assigned to it.
+    Checks the idleness of a session by the average utilization of compute devices.
     """
 
     name: ClassVar[str] = "utilization"
-    default_cpu_util_threshold: ClassVar[float] = 30.0
-    default_accelerator_util_threshold: ClassVar[float] = 10.0
-    default_initial_grace_period: ClassVar[float] = 300.0  # allow first 5 minutes to be idle
+    check_interval: float = 20.0
+
+    _config_iv = t.Dict(
+        {
+            t.Key("time-window", default="10m"): tx.TimeDuration(),
+            t.Key("initial-grace-period", default="5m"): tx.TimeDuration(),
+            t.Key("thresholds-check-operator", default="and"): t.String,
+            t.Key("resource-thresholds"): t.Dict(
+                {
+                    t.Key("cpu", default=None): t.Null | t.Dict({t.Key("average"): t.Float}),
+                    t.Key("mem", default=None): t.Null | t.Dict({t.Key("average"): t.Float}),
+                    t.Key("cuda_util", default=None): t.Null | t.Dict({t.Key("average"): t.Float}),
+                    t.Key("cuda_mem", default=None): t.Null | t.Dict({t.Key("average"): t.Float}),
+                }
+            ),
+        }
+    ).allow_extra("*")
+
+    resource_thresholds: MutableMapping[str, Any]
+    thresholds_check_operator: str
+    time_window: timedelta
+    _policy_cache: ContextVar[Dict[AccessKey, Optional[Mapping[str, Any]]]]
+    _evhandlers: List[EventHandler[None, AbstractEvent]]
 
-    async def populate_config(self, config: Mapping[str, Any]) -> None:
-        pass
+    async def __ainit__(self) -> None:
+        await super().__ainit__()
+        self._policy_cache = ContextVar("_policy_cache")
+        self._redis_stat = await aioredis.create_redis(
+            str(self._shared_config.get_redis_url(db=REDIS_STAT_DB))
+        )
 
-    async def check_session(self, session: Row, dbconn: SAConnection) -> bool:
-        # last_stat = session['last_stat']
-        # TODO: implement
-        return True
+    async def aclose(self) -> None:
+        self._redis_stat.close()
+        await self._redis_stat.wait_closed()
+        await super().aclose()
+
+    async def populate_config(self, raw_config: Mapping[str, Any]) -> None:
+        config = self._config_iv.check(raw_config)
+        config
```

Comment:
Delete this line.

Reply:
I implemented grace period, by that time the issue was resolved without needing the grace period. So that, I was deleting my grace period implementation. Currently, deleted all lines of grace period.

Reply:
I might get lost, but didn't we determined to introduce the `initial-grace-period` to support use cases for a longer initial-grace-period with a shorter time-window?

and, `config` in line 366 above should be deleted.

Reply:
Oh, I have misunderstood your comment before about introducing the grace period.
 I will add grace period configs and implementation soon.
 
 """Since the utilization checker does not terminate a session until the length of the collected data exceeds window-size. So, I'm not sure initial-grace-period is required. But, there may be use cases to set longer initial-grace-period with shorter window-size, so I don't have an objection to the introduction of the initial-grace-period. So, I'll mark this issue as resolved."""

Reply:
Reintroduced initial grace period